internal/worker/task.go

```diff
@@ -115,6 +115,7 @@ func (worker *Worker) runTask(
 func (worker *Worker) stopTask(taskID int64) {
 	if task, ok := worker.tasks[taskID]; ok {
 		task.cancel()
+		delete(worker.tasks, taskID)
```

Comment:
Which problem are you trying to solve with this?

I don't think we can currently just remove the task without waiting for it to terminate, because that would be a spurious task completion in logs:

https://github.com/cirruslabs/cirrus-cli/blob/0ebeb27313b1739911cd3bb5442c5542a301c19b/internal/worker/task.go#L151-L157

Plus, before we receive the task completion the task technically is still running.